------------------------------------------------------------
revno: 14074
revision-id: squid3@treenet.co.nz-20160817054829-rl7q49ysi40sj01i
parent: squid3@treenet.co.nz-20160817051037-p0kaj2iw2u4u8iqj
fixes bug: http://bugs.squid-cache.org/show_bug.cgi?id=3025
author: mkishi <mkishi@104.net>
committer: Amos Jeffries <squid3@treenet.co.nz>
branch nick: 3.5
timestamp: Wed 2016-08-17 17:48:29 +1200
message:
  Bug 3025: Proxy-Authenticate problem using ICAP server
------------------------------------------------------------
# Bazaar merge directive format 2 (Bazaar 0.90)
# revision_id: squid3@treenet.co.nz-20160817054829-rl7q49ysi40sj01i
# target_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# testament_sha1: f4eb1b35dc72bba74a398070900a0951257e547e
# timestamp: 2016-08-17 05:50:56 +0000
# source_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# base_revision_id: squid3@treenet.co.nz-20160817051037-\
#   p0kaj2iw2u4u8iqj
# 
# Begin patch
=== modified file 'src/client_side_reply.cc'
--- src/client_side_reply.cc	2016-04-01 06:15:31 +0000
+++ src/client_side_reply.cc	2016-08-17 05:48:29 +0000
@@ -1305,8 +1305,14 @@
 
     // if there is not configured a peer proxy with login=PASS or login=PASSTHRU option enabled
     // remove the Proxy-Authenticate header
-    if ( !request->peer_login || (strcmp(request->peer_login,"PASS") != 0 && strcmp(request->peer_login,"PASSTHRU") != 0))
-        reply->header.delById(HDR_PROXY_AUTHENTICATE);
+    if ( !request->peer_login || (strcmp(request->peer_login,"PASS") != 0 && strcmp(request->peer_login,"PASSTHRU") != 0)) {
+#if USE_ADAPTATION
+        // but allow adaptation services to authenticate clients
+        // via request satisfaction
+        if (!http->requestSatisfactionMode())
+#endif
+            reply->header.delById(HDR_PROXY_AUTHENTICATE);
+    }
 
     reply->header.removeHopByHopEntries();
 

=== modified file 'src/client_side_request.h'
--- src/client_side_request.h	2016-01-01 00:14:27 +0000
+++ src/client_side_request.h	2016-08-17 05:48:29 +0000
@@ -140,6 +140,7 @@
 
 public:
     void startAdaptation(const Adaptation::ServiceGroupPointer &g);
+    bool requestSatisfactionMode() const { return request_satisfaction_mode; }
 
     // private but exposed for ClientRequestContext
     void handleAdaptationFailure(int errDetail, bool bypassable = false);

